tests/tasks/setup/eks/awscli-pod-identity-association.yaml

```diff
@@ -0,0 +1,87 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: awscli-eks-pia-create
+  namespace: scalability
+spec:
+  description: |
+    Create an EKS Pod Identity Association for a given cluster.
+    This Task can be used to create an EKS Pod Identity Association for namespace default and service account default.
+  params:
+  - name: cluster-name
+    description: The name of the EKS cluster you want to create an EKS Pod Identity Association for.
+  - name: region
+    default: "us-west-2"
+    description: The region where the cluster is in.
+  - name: endpoint
+    default: ""
+  - name: namespace-prefix
+    default: "default"
+  - name: namespaces
+    default: "1"
+  workspaces:
+  - name: config
+    mountPath: /config/
+  stepTemplate:
+    env:
+    - name: KUBECONFIG
+      value: /config/kubeconfig
+  steps:
+  - name: write-kubeconfig
+    image: alpine/k8s:1.31.5
+    script: |
+      ENDPOINT_FLAG=""
+      if [ -n "$(params.endpoint)" ]; then
+        ENDPOINT_FLAG="--endpoint $(params.endpoint)"
+      fi
+      aws eks $ENDPOINT_FLAG update-kubeconfig --name $(params.cluster-name) --region $(params.region)
+  - name: create-pia
+    image: alpine/k8s:1.31.5
+    script: |
+      ENDPOINT_FLAG=""
+      if [ -n "$(params.endpoint)" ]; then
+        ENDPOINT_FLAG="--endpoint $(params.endpoint)"
+      fi
+
+      MANAGED_POLICY_ARN="arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess"
+      TRUST_POLICY_FILE="pia-trust-policy.json"
+      # create a trust policy json file
+      cat > $TRUST_POLICY_FILE <<EOF
+      {
+        "Version": "2012-10-17",
+        "Statement": [
+          {
+            "Effect": "Allow",
+            "Principal": {
+              "Service": "beta.pods.eks.aws.internal"
+            },
+            "Action": [
+              "sts:AssumeRole",
+              "sts:TagSession"
+            ]
+          }
+        ]
+      }
+      EOF
+      for i in $(seq 1 $(params.namespaces)); do
+        kubectl create namespace $(params.namespace-prefix)-$i
+
+        PIA_ROLE_NAME=$(params.cluster-name)-pia-role-$i
+        aws iam create-role --role-name $PIA_ROLE_NAME --assume-role-policy-document file://$TRUST_POLICY_FILE
+        aws iam attach-role-policy --role-name $PIA_ROLE_NAME --policy-arn $MANAGED_POLICY_ARN
+        PIA_ROLE_ARN=$(aws iam get-role --role-name $PIA_ROLE_NAME --query 'Role.Arn' --output text)
+        echo "$PIA_ROLE_ARN is created"
+
+        aws eks $ENDPOINT_FLAG --region $(params.region) create-pod-identity-association \
+        --cluster-name $(params.cluster-name) \
+        --namespace $(params.namespace-prefix)-$i \
+        --service-account default \
+        --role-arn $PIA_ROLE_ARN
+      done
+
+      aws eks $ENDPOINT_FLAG --region $(params.region) list-pod-identity-associations --cluster-name $(params.cluster-name) --query 'associations'
+
+      echo "waiting for 30 seconds..."
+      sleep 30
```

Comment:
is there a better way to know pod-identity-association went through than sleeping before we proceed ?

Reply:
We have synced up offline on this. The answer is no, currently there is not an interface on this